### PR TITLE
[Merged by Bors] - Add a UV sphere implementation

### PR DESCRIPTION
**Pull request by nside**
Added a UV sphere implementation

### CODE REVIEW

File: crates/bevy_render/src/mesh/shape/uvsphere.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            sectors: 36,
+            stacks: 18,
+        }
+    }
+}
+
+impl From<Uvsphere> for Mesh {
+    fn from(sphere: Uvsphere) -> Self {
+        // Largely inspired from http://www.songho.ca/opengl/gl_sphere.html
+        let PI: f32 = f32::acos(-1.);
```
Comment (jakobhellermann) [inline crates/bevy_render/src/mesh/shape/uvsphere.rs:30]: you can `use std::f32::consts::PI` instead

File: crates/bevy_render/src/mesh/shape/uvsphere.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+use crate::{
+    mesh::{Indices, Mesh},
+    pipeline::PrimitiveTopology,
+};
+
+/// A sphere made of sectors and stacks
+#[derive(Debug, Clone, Copy)]
+pub struct Uvsphere {
```
Comment (jakobhellermann) [inline crates/bevy_render/src/mesh/shape/uvsphere.rs:8]:
```suggestion
pub struct UVSphere {
```

**Comment by nside** [comment]
CleanShot 2021-04-11 at 11 59 21

File: crates/bevy_render/src/mesh/shape/uvsphere.rs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+/// A sphere made of sectors and stacks
+#[derive(Debug, Clone, Copy)]
+pub struct UVSphere {
+    /// The radius of the sphere.
+    pub radius: f32,
+    // Longitudinal sectors
+    pub sectors: usize,
+    // Latitudinal stacks
+    pub stacks: usize,
```
Comment (mockersf) [inline crates/bevy_render/src/mesh/shape/uvsphere.rs:15]:
```suggestion
    /// Longitudinal sectors
    pub sectors: usize,
    /// Latitudinal stacks
    pub stacks: usize,
```

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**